stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1724,7 +1724,7 @@ function normalize!(a::AbstractArray, p::Real=2)
     __normalize!(a, nrm)
 end
 
-@inline function __normalize!(a::AbstractArray, nrm::AbstractFloat)
+@inline function __normalize!(a::AbstractArray, nrm)
```

Comment:
Would it make more sense to just relax this to `nrm::Real` instead of `AbstractFloat`? I really don't think it makes sense for non-numeric types, and also it really doesn't make sense for non-real types to me either (unless someone knows of norms that are complex and computed this was)? Note that relaxing it to `Real` should let the `ForwardDiff.Dual` type work with it still.

Reply:
I agree, let's keep this one as `Real`. It makes it easier to read the code and might give more informative error messages if this method is unintentionally called.

Reply:
This makes sense, just fixed with the latest commit.